docs/proposals/device-state.md

```diff
@@ -0,0 +1,215 @@
+---
+title: Device state Design
+states: implementable
+authors:
+    - "@JiaweiGithub"
+approvers:
+creation-date: 2024-06-02
+last-updated: 2024-06-02
+---
+
+# Support device state message.
+
+## Motivation
+The device has its own states, such as online states, offline states, etc. It is very necessary to collect these stateses and report them to the cloud.
+### Goals
+- Mapper provides a common framework code to support collecting the states of the device itself.
+- After Edgecore receives the device states message, it updates the cache database device states message and 
+  send to cloud.
+- After receiving the device states message, Cloudcore updates the device's cr states information.
+
+### Non-goals
+- Mapper only provides a general framework and does not provide device states collection code.
+
+### Device state design
+<img src="../images/device-crd/device-state.png">
+
+### Mapper design
+- In the dmi module, add a grpc interface for ReportDeviceState.
+- Call getDeviceStates based on goroutine in Mapper's DevStart.
+- GetDeviceStates regularly obtains the device states, and then sends it to edgecore through the ReportDeviceState 
+  interface.
+
+```golang
+func (gs *DeviceStates) Run() {
+	states, error := gs.Client.GetDeviceStates()
+	if error != nil {
+		klog.Errorf("GetDeviceStates failed: %v", error)
+		return
+	}
+
+	statesRequest := &dmiapi.ReportDeviceStatesRequest{
+		DeviceName:      gs.DeviceName,
+		State:           states,
+		DeviceNamespace: gs.DeviceNamespace,
+	}
+
+	log.Printf("send statesRequest", statesRequest.DeviceName, statesRequest.State)
+	if err := grpcclient.ReportDeviceStates(statesRequest); err != nil {
+		klog.Errorf("fail to report device states of %s with err: %+v", gs.DeviceName, err)
+	}
```

Comment:
I don’t seem to see how to report regularly from the code. Maybe you can use pseudocode to illustrate the overall functionality